python/utilities.py

```diff
@@ -187,44 +188,74 @@ def isValidSolution(solution, referenceLatlon, state):
             break
 
         rospy.logwarn("Attempting to rerun with longer runtime: {} seconds".format(runtimeSeconds))
-        # TODO: Incorporate globalWindSpeed into pathfinding?
         solution = plan(runtimeSeconds, "RRTStar", 'WeightedLengthAndClearanceCombo', globalWindDirectionDegrees, dimensions, start, goal, obstacles)
-
         if isValidSolution(solution, referenceLatlon, state):
             solutions.append(solution)
         else:
             invalidSolutions.append(solution)
 
-    # Choose best valid solution. If no valid solutions found, use the best invalid one.
-    if len(solutions) > 0:
-        solution = min(solutions, key=lambda x: x.getSolutionPath().cost(x.getOptimizationObjective()).value())
+    # If no valid solutions found, use the best invalid one. Do not perform any path simplifying on invalid paths.
+    if len(solutions) == 0:
+        bestSolution = min(invalidSolutions, key=lambda x: x.getSolutionPath().cost(x.getOptimizationObjective()).value())
+        bestSolutionPath = bestSolution.getSolutionPath()
+        minCost = bestSolutionPath.cost(bestSolution.getOptimizationObjective()).value()
     else:
-        solution = min(invalidSolutions, key=lambda x: x.getSolutionPath().cost(x.getOptimizationObjective()).value())
+
+        # Find path with minimum cost. Can be either simplified or unsimplified path.
+        # Need to recheck that simplified paths are valid before using
+        minCost = sys.maxsize
+        bestSolution = None
+        bestSolutionPath = None
+        for solution in solutions:
+            # Check unsimplified path
+            unsimplifiedPath = og.PathGeometric(solution.getSolutionPath())
+            unsimplifiedCost = unsimplifiedPath.cost(solution.getOptimizationObjective()).value()
+            if unsimplifiedCost < minCost:
+                bestSolution = solution
+                bestSolutionPath = unsimplifiedPath
+                minCost = unsimplifiedCost
+
+            # Check simplified path
+            solution.simplifySolution()
```

Comment:
@patrick-5546 this is the simplifying thing. Should probably check when this part was removed and why.